src/NuGet.Services.Metadata.Catalog.Monitoring/LoggingUtility.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Logging;
+using System;
+using System.Linq;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring
+{
+    /// <summary>
+    /// <see cref="Common.ILogger"/> wrapper for <see cref="Microsoft.Extensions.Logging.ILogger"/>.
+    /// </summary>
+    public class CommonLogger : Common.ILogger
+    {
+        public CommonLogger(ILogger logger)
+        {
+            InternalLogger = logger ?? throw new ArgumentNullException(nameof(ILogger));
+        }
+
+        public ILogger InternalLogger { get; private set; }
+
+        public void LogDebug(string data)
+        {
+            InternalLogger.LogDebug(data);
+        }
+
+        public void LogVerbose(string data)
+        {
+            InternalLogger.LogInformation(data);
+        }
+
+        public void LogInformation(string data)
+        {
+            InternalLogger.LogInformation(data);
+        }
+
+        public void LogMinimal(string data)
+        {
+            InternalLogger.LogInformation(data);
+        }
+
+        public void LogWarning(string data)
+        {
+            InternalLogger.LogWarning(data);
+        }
+
+        public void LogError(string data)
+        {
+            InternalLogger.LogError(data);
+        }
+
+        public void LogInformationSummary(string data)
+        {
+            InternalLogger.LogInformation(data);
+        }
+
+        public void LogErrorSummary(string data)
+        {
+            InternalLogger.LogError(data);
+        }
+    }
+
+    public static class ILoggerExtensions
+    {
+        /// <summary>
+        /// Wraps <paramref name="logger"/> as a <see cref="Common.ILogger"/>.
+        /// </summary>
+        public static Common.ILogger AsCommon(this ILogger logger)
+        {
+            return new CommonLogger(logger);
+        }
+    }
+
+    public static class ILoggerFactoryExtensions
+    {
+        /// <summary>
+        /// Replacement for <see cref="LoggerFactoryExtensions.CreateLogger(ILoggerFactory, Type)"/> that creates an <see cref="ILogger{TCategoryName}"/>.
+        /// </summary>
+        public static ILogger CreateTypedLogger(this ILoggerFactory loggerFactory, Type type)
```

Comment:
Okay, this seems a little off that we can't just use type parameters all the way. I understand we have this "initialize by type name" pattern in one of your factories. The result of that design decision is loss of compile time goodness, like being able to use a type parameters here. Seems like you are doing the work that something in Microsoft.Extensions.* should be doing.